src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,13 +3,91 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private final Pair<K,V>[] storageArray;
+    private int size;
+
+    public StorageImpl() {
+        storageArray = new Pair[MAX_SIZE];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
+
+        for (int i = 0; i <= size; i++) {
+            if (i == size) {
+                storageArray[i] = Pair.of(key, value);
+                size++;
+                return;
+            }
+            if (storageArray[i].firstVariable == key
+                    || storageArray[i].equals(key)
+                    && storageArray[i].firstVariable != null) {
```

Comment:
```suggestion
                    || storageArray[i].firstVariable != null && storageArray[i].firstVariable.equals(key)) {
```
can you explain why this fix is needed?

Reply:
this fix is needed to not create new Pair object when key is existing. This fix will just change second pair value of existing key
Also, understood that null checking is not necessary because key of existing pair cannot be null
fixed